### DISCUSSION TITLE
store operation: single ancestor or multiple nodes?

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#520] [Q&A] [answered]
Hi all;
given [this](https://static.javatpoint.com/ds/images/types-of-tree1.png) object graph
assumning I need to change, in one logical operation nodes: E,H & Q
what is the way I need to store things for this opetation:
1. to to the ancestor common parent and store it using store(A)
2. store all 3 changed nodes using storeAll(E,H,Q)
will both work? if they both work will they perform the same?
thanks.

**Comment by hg-ms** [comment, marked as answer]
Hi,
The storeAll(E,H,Q) is the recommended solution to update several objects at once. It will apply a [Lazy-Storer]( https://docs.microstream.one/manual/storage/storing-data/lazy-eager-full.html) to all three objects. The storeAll method will either persist all supplied instance or non and throw an exception in case of error.

Using store(A) would work too if using an [Eager-Storer]( https://docs.microstream.one/manual/storage/storing-data/lazy-eager-full.html#_usage) but this will store everything reachable by references from “A” and therefore cause a lot of unnecessary overhead. In your case that would write out the whole graph again starting at “A”
